src/event-handler/event-channel-manager.ts

```diff
@@ -0,0 +1,125 @@
+import Ain from '../ain';
+import EventHandler from './index';
+import { WebSocket } from 'ws';
+import {
+  EventChannelMessageTypes,
+  EventChannelMessage,
+  BlockchainEventTypes,
+} from '../types';
+import EventFilter from './event-filter';
+
+export default class EventChannelManager {
+  private readonly _ain: Ain;
+  private readonly _eventHandler: EventHandler;
+  private _wsClient?: WebSocket;
+  private _endpointUrl?: string;
+  private _isConnected: boolean;
+
+  constructor(ain: Ain, eventHandler) {
+    this._ain = ain;
+    this._eventHandler = eventHandler;
+    this._wsClient = undefined;
+    this._endpointUrl = undefined;
+    this._isConnected = false;
+  }
+
+  get isConnected(): boolean {
+    return this._isConnected;
+  }
+
+  connect() {
+    return new Promise(async (resolve, reject) => {
+      const eventHandlerNetworkInfo = await this._ain.net.getEventHandlerNetworkInfo();
+      const url = eventHandlerNetworkInfo.url;
+      if (!url) {
+        reject(new Error(`Can't get url from eventHandlerNetworkInfo ` +
+            `(${JSON.stringify(eventHandlerNetworkInfo, null, 2)}`));
+      }
+      this._endpointUrl = url;
+      this._wsClient = new WebSocket(url);
+      this._wsClient.on('message', (message) => {
+        this.handleMessage(message);
+      });
+      this._wsClient.on('open', () => {
+        this._isConnected = true;
+        resolve();
```

Comment:
what about setting up a timeout?

Reply:
Looks nice! I added handshakeTimeout option to `connect()` function and I researched wsClient.on('error') can catch timeout error. Done